internal/api/logs/v1/http.go

```diff
@@ -121,6 +121,35 @@ func NewHandler(read, write *url.URL, opts ...HandlerOption) http.Handler {
 		})
 	}
 
+	if tail != nil {
+		var tailRead http.Handler
+		{
+			tail.Path = path.Join(tail.Path, lokiUpstreamPrefix)
+			middlewares := proxy.Middlewares(
+				proxy.MiddlewareSetUpstream(tail),
+				proxy.MiddlewareLogger(c.logger),
+				proxy.MiddlewareMetrics(c.registry, prometheus.Labels{"proxy": "logsv1-tail"}),
+			)
+
+			tailRead = &httputil.ReverseProxy{
+				Director: middlewares,
+				ErrorLog: proxy.Logger(c.logger),
+				Transport: &http.Transport{
+					DialContext: (&net.Dialer{
+						Timeout: ReadTimeout,
+					}).DialContext,
+				},
+			}
+		}
+		r.Group(func(r chi.Router) {
+			r.Use(c.readMiddlewares...)
```

Comment:
Do you think this is sufficient to use the authorization middleware enforcing the `read` permission? Or do we need a new `tail` or `watch` permission? In other words, is tailing any different from a security perspective than reading? e.g. more load on server?

Reply:
@squat IMHO there is no need to have explicit permissions for tailing the logs. The read endpoint can do the same thing without the capability to stream newer logs.

Besides permissions, do you think it is better to call this endpoint watch instead of tail? Tail is closer from a terminology standpoint to Loki and logs, but watch is closer to k8s. WDYT?

Reply:
I don't have a strong opinion about the endpoint name. I would have said `watch` so that it could be generic, but we likely won't have analogous endpoints for other signals, i.e. there won't be a tail/watch for metrics/traces/profiles etc so we can make a one-off `tail` endpoint if we want

Reply:
Ok! Let's keep it tail then.